module/zfs/arc.c

```diff
@@ -9337,26 +9337,27 @@ l2arc_apply_transforms(spa_t *spa, arc_buf_hdr_t *hdr, uint64_t asize,
 	}
 
 	if (compress != ZIO_COMPRESS_OFF && !HDR_COMPRESSION_ENABLED(hdr)) {
-		cabd = abd_alloc_for_io(asize, ismd);
-		tmp = abd_borrow_buf(cabd, asize);
+		cabd = abd_alloc_for_io(size, ismd);
+		tmp = abd_borrow_buf(cabd, size);
 
 		psize = zio_compress_data(compress, to_write, tmp, size,
 		    hdr->b_complevel);
```

Comment:
Would you mind adding a comment saying that there are cases of `size > asize` (e.g encryption) and we have to allocate `tmp` and `cabd` with size instead of asize.

Reply:
I have no problem with adding the comment that it can happen, but I'd want to have much more detailed examples if I'm going to include anything beyond "it can happen", I think - the last thing I'd like to do is suggest someone look in a certain part of the code for the cause and be incorrect.

Reply:
However, I am fine with not providing detailed examples of why (`asize < size`).

Reply:
There we are. How's that?